tests/lib/prepare-restore.sh

```diff
@@ -312,6 +312,16 @@ prepare_project() {
     # remove any packages that are marked for auto removal before running any tests
     distro_auto_remove_packages
 
+    if os.query is-amazon-linux 2023; then
+        # perform system upgrade to the latest release
+        if [[ "$SPREAD_REBOOT" == 0 ]]; then
+            if distro_upgrade | MATCH "reboot"; then
```

Comment:
We can do that in the update that we run weekly, so we don't need to do it in any pr, what do you prefer?